packages/cli/src/credentials/credentials.service.ts

```diff
@@ -300,16 +300,23 @@ export class CredentialsService {
 		for (const dataKey of Object.keys(copiedData)) {
 			// The frontend only cares that this value isn't falsy.
 			if (dataKey === 'oauthTokenData') {
-				copiedData[dataKey] = CREDENTIAL_BLANKING_VALUE;
+				if (copiedData[dataKey].toString().length > 0) {
+					copiedData[dataKey] = CREDENTIAL_BLANKING_VALUE;
+				} else {
+					copiedData[dataKey] = CREDENTIAL_EMPTY_VALUE;
+				}
 				continue;
 			}
 			const prop = properties.find((v) => v.name === dataKey);
 			if (!prop) {
 				continue;
 			}
 			if (prop.typeOptions?.password) {
-				// eslint-disable-next-line @typescript-eslint/no-unsafe-assignment, @typescript-eslint/no-unsafe-member-access
-				copiedData[dataKey] = CREDENTIAL_BLANKING_VALUE;
+				if (copiedData[dataKey].toString().length > 0) {
+					copiedData[dataKey] = CREDENTIAL_BLANKING_VALUE;
+				} else {
+					copiedData[dataKey] = CREDENTIAL_EMPTY_VALUE;
```

Comment:
Why don't we simply use an empty string value? We could send it to FE and it would simply display as empty to the user. One details is that if the user clicks to save, then the property would be removed from the credential, but that's ok since it wouldn't work anyway.

I think it's complicating more by adding this new possible `empty_value` constant WDYT?

Reply:
we can do that. thing is, blank strings will not be sent to the frontend (i assume we're checking for truthy somewhere) so the frontend simply puts in the default, which I presume results in the same empty string. In the opposite direction, the empty string field is also not sent to the backend in a PATCH request, thus resulting in whatever is default being put into that field.
so the difference is basically: with the explicit empty alias value, we circumvent the field not being sent, with the extra replacement being necessary on both ends. without, we have to rely on the default values being empty.

Reply:
Gotcha - I don't think that this would break something, but it's good to be on the safe side. Usually default values are empty but that's not a rule, so we can't rely on it.